index.js

```
@@ -48,50 +48,341 @@ const player = {
     { id: 5, name: 'Israeli', songs: [4, 5] },
   ],
   playSong(song) {
-    console.log(/* your code here */)
+    console.log("Playing " + song.title + " from " + song.album + " by " + song.artist + " | " + durationFormat(song.duration) + ".")
   },
 }
 
 function playSong(id) {
-  // your code here
+  let songIndex = player.songs.findIndex(i => i.id === id) // finds the index of the id in the songs array
+  if (songIndex !== -1){
+    player.playSong(player.songs[songIndex])
+  }
+  else{
+    throw Error("Song Doesn't Exist");
+  }    
+  
 }
 
 function removeSong(id) {
-  // your code here
+  let songIndex = player.songs.findIndex(i => i.id === id)
+  if (songIndex !== -1){
+    player.songs.splice(songIndex, 1)
+  }
+  else {
+    throw Error("Song Doesn't Exist");
+  }
+  let playlistIndex = -1;
+  let removed = false; // Indicates if the song was removed.
+  for (let i = 0; i<player.playlists.length; i++){ // Checks every playlist if it has the song.
+    playlistIndex = player.playlists[i].songs.indexOf(id);
+    if (player.playlists[i].songs.indexOf(id) !== -1){
+      player.playlists[i].songs.splice(playlistIndex, 1);
+      removed = true; 
+    }
+  }
+  if (removed === false){
```

What if there's a song that doesn't belong to any playlist? Do you still want to throw an error?